RELEASE.md

```diff
@@ -0,0 +1,6 @@
+Release type: minor
+
+New `Lazy` that will replace `LazyType` in the future. It is aliased to
```

Comment:
sorry I just released we are renaming LazyType to Lazy, is there any reason for that? 

Could we keep the same name?

Reply:
You want to keep `LazyType` or just `Lazy`?

We can keep just one! =P. Since I created the PR as a proof of concepot and didn't want to make any radical changes if you guys didn't like it, I decided to create a new `Lazy` for that. But it is up to you :)

Reply:
we should keep one, I think the name `LazyType` might be misleading since it is similar to `typings.Type`, so maybe `Lazy` is better /cc @BryceBeagle what do you think?

Reply:
Hey sorry for the delayed response -- just seeing this now! I think the potential for confusion with `Type` could occur, but I'm not sure it would be that common. Another option could be `LazyVar` (similar to `ClassVar` and `TypeVar`) but I don't really have strong preferences

Reply:
For me it is up to you @patrick91 and @BryceBeagle to call it `Lazy`, `LazyVar`, or just keep the current `LazyType` and define it in the `else` part of a `if TYPE_CHECKING`.

Reply:
Let's go with `LazyVar`! 😊